File: web/src/main/java/no/nav/modiapersonoversikt/api/domain/Oppgave.java
package no.nav.modiapersonoversikt.api.domain;

import java.io.Serializable;

public class Oppgave implements Serializable {

    public final String oppgaveId, fnr, henvendelseId;
    public final boolean erSTOOppgave;
    public String svarHenvendelseId;

    public Oppgave(String oppgaveId, String fnr, String henvendelseId, boolean erSTOOppgave) {
        this.oppgaveId = oppgaveId;
        this.fnr = fnr;
        this.henvendelseId = henvendelseId;
        this.erSTOOppgave = erSTOOppgave;
    }

    public Oppgave withSvarHenvendelseId(String svarHenvendelseId) {
        this.svarHenvendelseId = svarHenvendelseId;
        return this;
    }

}

File: web/src/main/java/no/nav/modiapersonoversikt/service/oppgavebehandling/Oppgave.kt
package no.nav.modiapersonoversikt.service.oppgavebehandling

import java.time.LocalDate
import java.time.LocalDateTime

data class Oppgave(
    val oppgaveId: String,
    val fnr: String,
    val henvendelseId: String?,
    val erSTOOppgave: Boolean,
    val tildeltEnhetsnr: String,
    val fristFerdigstillelse: LocalDate? = null,
    val tema: String,
    val temagruppe: String? = null,
    val oppgavetype: String,
    val prioritet: String,
    val status: String,
    val aktivDato: LocalDate,
    val endretAvEnhetsnr: String? = null,
    val opprettetAvEnhetsnr: String? = null,
    val saksreferanse: String? = null,
    val beskrivelse: String? = null,
    val opprettetTidspunkt: LocalDateTime? = null,
)
